sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -362,7 +371,24 @@ public Boolean getBoolean(int idx) {
   @Nullable
   public <T> T getLogicalTypeValue(int idx, Class<T> clazz) {
     LogicalType logicalType = checkNotNull(getSchema().getField(idx).getType().getLogicalType());
-    return (T) logicalType.toInputType(getValue(idx));
+    return (T) getValue(idx);
+  }
+
+  /**
+   * Returns the Logical Type base type for this field. If there are multiple nested logical types,
+   * they are all resolved to the first non-logical type. {@link IllegalStateException} is thrown if
+   * schema doesn't match.
+   */
+  @Nullable
+  public <T> T getLogicalTypeBaseValue(int idx, Class<T> clazz) {
+    Object value = getValue(idx);
+    FieldType fieldType = getSchema().getField(idx).getType();
+    checkArgument(fieldType.getTypeName().isLogicalType());
+    while (fieldType.getTypeName().isLogicalType()) {
+      value = fieldType.getLogicalType().toBaseType(value);
```

Comment:
Should `toBaseType` handle null values for nullable fields or not? Can we assume that `toBaseType(null) == null`?

Reply:
good catch, we should.